securing-communication/index.md

```diff
@@ -0,0 +1,877 @@
+
+That same key is used for both encrypting and decrypting data.
+AES (*Advanced Encryption Standard*) is the standard symmetric encryption algorithm.
+The main benefit of symmetric encryption algorithms is their relative simplicity and speed: they are easy to implement and are fast, with the option of having hardware support.
+The downside is related to the shared key.
+If the key is captured by an attacker or if it is lost by one of the ends, confidentiality is compromised.
+
+So, the goal is that each connection should use a temporary shared key.
+After the connection ends, the shared key is discarded.
+A new connection will generate a new key.
+Of course, that shared key must be known only by the two ends.
+For this two happen, key exchange algorithms, such as [Diffie Hellman](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) are used.
+
+Diffie-Hellman (often abbreviated as DH) is based on public-key encryption.
+In short we use public-key encryption to set up a temporary shared key for the actual communication.
+
+Public-key encryption, as its name implies, relies on a pair of private and public keys that are connected mathematically.
+RSA (*Rivest Shamir Adleman* - named after its inventors) is the classical public key cryptographic algorithm.
+The private key is generated as a random set of bytes and the public key is generated from it, via a mathematical algorithm.
+The private key is only available to the owner, whereas the public key is available to everyone.
+Anyone can encrypt a message using the public key, but only the owner can decrypt the message using the private key, as shown below.
+
+![public-key encryption](assets/public-key-encryption.svg)
+
+Because of this, public-key encryption is considered more secure than symmetric encryption, as it doesn't require the passing of a shared key between parties, an act that may be intercepted.
+At the same time, public-key encryption is much slower than symmetric encryption.
+Because of this, public-key encryption is only used to set up an initial session and enable a key exchange algorithm (such as Diffie-Hellman) to generate a temporary session-specific shared key.
+
+### Identity Management. Certificates
+
+A public-private key pair is not only used for encryption.
+It's also used for identity management.
+
+Identity management means making sure a given entity is who they claim they are.
+In HTTPS that means that if we connect to `google.com` we are offered guarantees that the server we connect to is indeed `google.com`.
+Otherwise, another server could impersonate the target server and capture all traffic.
+
+Identity management relies on signing and verifying messages using public-private keys.
+The private key is used to sign a message.
+The signed message is provided publicly.
+Then, the public key is used to verify the message.
+This is called a **digital signature**, as shown below:
+
+![digital signature](assets/digital-signature.svg)
```

Comment:
This image needs to be translated into English.

Reply:
Yes, I know, will keep it like this for now.